pkg/cmd/pulumi/up.go

```diff
@@ -299,10 +317,11 @@ func newUpCmd() *cobra.Command {
 		}
 
 		opts.Engine = engine.UpdateOptions{
-			LocalPolicyPacks: engine.MakeLocalPolicyPacks(policyPackPaths, policyPackConfigPaths),
-			Parallel:         parallel,
-			Debug:            debug,
-			Refresh:          refreshOption,
+			LocalPolicyPacks:  engine.MakeLocalPolicyPacks(policyPackPaths, policyPackConfigPaths),
+			Parallel:          parallel,
+			Debug:             debug,
+			Refresh:           refreshOption,
+			ExperimentalPlans: hasExperimentalCommands(),
```

Comment:
Should this check for `planFilePath != ""`?

Reply:
Nope, even if we don't have a plan this turns on the generation of plans so that the inline preview for up generates a plan and passes it on to the actual update operation. That is we actually *always* use plans even if no plan file is given.